shared/slices/slice_generic.go

```diff
@@ -0,0 +1,308 @@
+package slices
```

Comment:
Can we add a package level go doc comment here explaining: Package slices contains utility functions for dealing with slices, inferring underlying types using the reflect package to work for as many types as possible.

Reply:
Will be changing the names and adding the go doc for this